content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md

```diff
@@ -0,0 +1,335 @@
+        node<T> prev = new node<>(null);
+        prev.next = head;
+        node<T> next = head.next;
+        node<T> newtemp = head;
+        boolean exists = false;
+        if (head.data == key) {
+            head = head.next;
+            exists = true;
+        }
+        while (newtemp.next != null) {
+            if (String.valueOf(newtemp.data).equals(
+                    String.valueOf(key))) {
+                prev.next = next;
+                exists = true;
+                break;
+            }
+            prev = newtemp;
+            newtemp = newtemp.next;
+            next = newtemp.next;
+        }
+        if (exists == false
+            && String.valueOf(newtemp.data).equals(
+                String.valueOf(key))) {
+            prev.next = null;
+            exists = true;
+        }
+        if (exists) {
+            newlength--;
+        }
+        else {
+            System.out.println(
+                "The linked list does not contain the given value.");
+        }
+    }
+    void clear()
+    {
+        head = null;
+        newlength = 0;
+    }
+  // It is the value that determines if the value to be erased is already there or not.
+    boolean empty()
+    {
+        if (head == null) {
+            return true;
+        }
+        return false;
+    }
+  // Returning the length of Linked List
+    int newlength() { return this.newlength; }
+// method
+    public String toString()
+    {
+        String S = "{ ";
+        node<T> X = head;
+        if (X == null)
+            return S + " }";
+        while (X.next != null) {
+            S += String.valueOf(X.data) + " -> ";
+            X = X.next;
+        }
+        S += String.valueOf(X.data);
+        return S + " }";
+    }
+}
+}}
+```
+
+Go ahead and run the code [here](https://replit.com/@Calvin49/2Implementing-a-Generic-Singly-linked-list-Data-Structure-in)
+
+Output:
+
+```bash
+ javac -classpath .:/run_dir/junit-4.12.jar:target/dependency/* -d . Main.java
+ java -classpath .:/run_dir/junit-4.12.jar:target/dependency/* Main
+>
+```
+
+> For the sake of simplicity, we use a dummy node to represent the prior one in singly-linked lists, since going backwards is impossible.
+> To represent the node before the current one, we create a dummy node. Because there isn't a previous node for the head node, it is assigned the null value.
+
+Here is an example:
+
+```Java
+// Node that has no data in it
+        node<T> prev = new node<>(null);
+        // head-pointing dummy node
+        prev.next = head;
+        // The node after the current node that points forward
+        node<T> next = head.next;
+```
+
+Take advantage of Generics' SinglyLinkedList. `Node<T>` will be the type of node variable. Now that we have covered the basics, let's get down to business by implementing the methods below:
+- Last but not least, we'll use add(T data).
+- Add the element at the top of the list.
+- Add an element to a specific location, use add(T data, int index).

```

Comment:
@LynnMumo, The paragraph here doesn't seem to flow very, especially how you started listing the methods "Last but not least. Kindly rephrase it in a more comprehensive way.